internal/core/algorithm/ngt/ngt.go

```diff
@@ -210,12 +210,22 @@ func gen(isLoad bool, opts ...Option) (NGT, error) {
 }
 
 func (n *ngt) setup() error {
-	n.ebuf = C.ngt_create_error_object()
+	n.epool = sync.Pool{
+		New: func() interface{} {
+			return C.ngt_create_error_object()
+		},
+	}
+
+	for i := 0; i < 20; i++ {
+		n.PutErrorBuffer(C.ngt_create_error_object())
+	}
 
-	n.prop = C.ngt_create_property(n.ebuf)
+	ebuf := n.GetErrorBuffer()
+	n.prop = C.ngt_create_property(ebuf)
```

Comment:
can we also cache this? 🤔

Reply:
I already cached this buffer please read GetErrorBuffer and PutErrorBuffer codes.